dockstore-webservice/src/main/java/io/dockstore/webservice/resources/SourceFileResource.java

```diff
@@ -0,0 +1,96 @@
+package io.dockstore.webservice.resources;
+
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import com.codahale.metrics.annotation.Timed;
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.core.SourceFile;
+import io.dockstore.webservice.core.Tag;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.core.WorkflowVersion;
+import io.dockstore.webservice.jdbi.TagDAO;
+import io.dockstore.webservice.jdbi.WorkflowVersionDAO;
+import io.dropwizard.auth.Auth;
+import io.dropwizard.hibernate.UnitOfWork;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.Authorization;
+import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
+import io.swagger.v3.oas.annotations.security.SecurityScheme;
+import io.swagger.v3.oas.annotations.security.SecuritySchemes;
+import org.apache.http.HttpStatus;
+import org.hibernate.SessionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.dockstore.webservice.Constants.JWT_SECURITY_DEFINITION_NAME;
+
+@Path("/versions")
```

Comment:
I think this should be hierarchical.

```
/workflows/<workflowid>/versions/<versionid>/sourcefiles
```

And the corresponding one for tags.

1. The hierarchy is a common RESTful pattern
2. Your way relies on ids of workflow versions and tags always being unique, which they are now, but who knows, some day that won't be.

See if anybody else agrees with me first though.

If you do it that way, then you should probably add the endpoints to WorkflowResource and DockerRepoTagResource.

Reply:
Ya, this was my first thought too.  The downside is you'd have two endpoints instead of one.  you'd also need the entry id

Reply:
changed to two endpoints